### PR TITLE
Add unit tests for `block_has_support`

### PR DESCRIPTION
**Pull request by ntsekouras**
Trac ticket: https://core.trac.wordpress.org/ticket/53257

This PR just adds some unit tests for `block_has_support` function.

What I wanted to verify though is if this is the expected behavior in the case we have declared `supports` for a key with an `empty array`. Should this return true as it does now?

Example test that passes with an empty array:
```
public function test_block_has_support_empty_array() {
	$this->registry->register(
		'core/example',
		array(
			'supports' => array(
				'align' => array(),
			),
		)
	);
	$block_type = $this->registry->get_registered( 'core/example' );
	$align_support = block_has_support( $block_type, array( 'align' ) );
	$this->assertSame( $align_support, true );
}
```

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/blocks/block.php [hunk 1]
```diff
@@ -567,4 +567,65 @@ public function test_build_query_vars_from_query_block_page_with_offset() {
 			)
 		);
 	}
+
+	/**
+	 * @ticket 53257
```
Comment (ntsekouras) [inline tests/phpunit/tests/blocks/block.php:572]: Should I change the ticket with the one introduced this function here as well? @gziolo

File: tests/phpunit/tests/blocks/block.php [hunk 2]
```diff
@@ -567,4 +567,65 @@ public function test_build_query_vars_from_query_block_page_with_offset() {
...
+					'color'    => array(
+						'link'     => true,
+						'gradient' => false,
+					),
+				),
+			)
+		);
+		$block_type    = $this->registry->get_registered( 'core/example' );
+		$align_support = block_has_support( $block_type, array( 'align' ) );
+		$this->assertSame( $align_support, true );
```
Comment (gziolo) [inline tests/phpunit/tests/blocks/block.php:590]: There is `assertFalse` and `assertTrue` in PHPUnit :)

PHPUnit has also `@dataProvider` functionality that would be another alternative. It allows you to run the same test with different test data which in practice

File: tests/phpunit/tests/blocks/block.php (hunk 1 above)
Comment (gziolo) [reply tests/phpunit/tests/blocks/block.php:572]: Yes, that would be a better fit.

**Comment by gziolo** [comment]
> What I wanted to verify though is if this is the expected behavior in the case we have declared supports for a key with an empty array. Should this return true as it does now?

You can file an issue to discuss it separately. It's an edge case that you don't see in a real life. I believe it should return `false` but it's also could be considered wrong input data like providing a number.

File: tests/phpunit/tests/blocks/block.php (hunk 2 above)
Comment (ntsekouras) [reply tests/phpunit/tests/blocks/block.php]: > assertFalse

That would make perfect sense and should have thought of that :) - made the changes.

**Comment by gziolo** [comment]
Committed in https://core.trac.wordpress.org/changeset/50955.

**Pull request closed without merging**